### PR TITLE
fix: bug with service user id vs lms user id

### PR DESCRIPTION
**Pull request by robrap**
**Description:**

- Fixed bug for successful JWTs where the JWT user id was still
using the service user id, rather than the LMS user id, so comparison
against the LMS user id would fail.
- As part of the bug fix, the custom attribute
``failed_jwt_cookie_user_id`` was renamed to
``jwt_cookie_lms_user_id``, and will be set for all JWT cookies.
Since this is only a breaking change for recently added monitoring,
this won't be versioned as a breaking change.

This is part of:
https://github.com/edx/edx-arch-experiments/issues/429

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bump if needed
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: edx_rest_framework_extensions/auth/jwt/authentication.py [hunk 1]
```diff
@@ -371,6 +332,34 @@ def _is_jwt_cookie_and_session_user_mismatch(self, request, jwt_user_id):
...
+        """
+        Returns the LMS user id from the JWT cookie, or None if not found
+
+        Notes:
+        - Must only be called in the case of a JWT cookie.
+        - Also provides monitoring details for mismatches.
+        """
+        try:
+            cookie_token = JSONWebTokenAuthentication.get_token_from_cookies(request.COOKIES)
+            invalid_decoded_jwt = unsafe_jwt_decode_handler(cookie_token)
```
Comment (robrap) [inline edx_rest_framework_extensions/auth/jwt/authentication.py:345]: **Author note**:
1. Reviewing by commit may be simpler, because this method was moved in a separate commit.
2. We will now be decoding all JWT cookies, which hopefully won't be a performance issue since we aren't even verifying the signature or anything.

Review can mark as resolved once read.

File: edx_rest_framework_extensions/auth/jwt/authentication.py [hunk 2]
```diff
@@ -371,6 +332,34 @@ def _is_jwt_cookie_and_session_user_mismatch(self, request, jwt_user_id):
 
         return True
 
+    def _get_and_monitor_jwt_cookie_lms_user_id(self, request):
+        """
+        Returns the LMS user id from the JWT cookie, or None if not found
+
+        Notes:
+        - Must only be called in the case of a JWT cookie.
```
Comment (timmc-edx) [inline edx_rest_framework_extensions/auth/jwt/authentication.py:340]: This sentence used to end with "failure", but the removal makes it unclear to me. Maybe this?

```suggestion
        - Must only be called when a JWT cookie is present.
```

**Review by timmc-edx — APPROVED** [review]
One doc nit, but otherwise looks good.

File: edx_rest_framework_extensions/auth/jwt/authentication.py (hunk 2 above)
Comment (robrap) [reply edx_rest_framework_extensions/auth/jwt/authentication.py]: I deleted the line instead. Thanks.
